### PR TITLE
Add remove ifaces from network

### PR DESCRIPTION
**Pull request by sajith**
Issue is #124.

### CODE REVIEW

File: fabrictestbed_extensions/fablib/node.py [hunk 1]
```diff
@@ -258,24 +270,11 @@ def toDict(self, skip=[]):
...
 
-    def get_template_contextXXX(self):
-        context = {}
-
-        context["node"] = self.toDict(skip=["ssh_command"])
-        context["slice"] = self.get_slice().toDict()
-        context["config"] = self.get_fablib_manager().get_config()
-
-        context["components"] = {}
+    def generate_template_context(self):
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/node.py:273]: What is this method for?

File: fabrictestbed_extensions/fablib/node.py [hunk 2]
```diff
@@ -1126,16 +1133,16 @@ def execute_thread(
...
+        retry: int = 3,
+        retry_interval: int = 10,
+        username: str = None,
+        private_key_file: str = None,
+        private_key_passphrase: str = None,
+        quiet: bool = False,
+        read_timeout: int = 10,
         timeout=None,
-        output_file=None,
+        output_file: str = None,
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/node.py:1145]: Maybe `typing.Union[str, os.PathLike]` (or maybe `pathlib.Path`) is more appropriate for file paths?

File: fabrictestbed_extensions/fablib/node.py [hunk 3]
```diff
@@ -2417,3 +2431,220 @@ def add_storage(self, name: str, auto_mount: bool = False) -> Component:
         :rtype: Component
         """
         return Component.new_storage(node=self, name=name, auto_mount=auto_mount)
+
+    def get_fim(self):
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/node.py:2435]: More places to add docstrings in this module too.

File: fabrictestbed_extensions/fablib/resources.py [hunk 4]
```diff
@@ -195,6 +196,18 @@ def get_topology_site(self, site_name: str) -> str:
             logging.warning(f"Failed to get site {site_name}")
             return ""
 
+    def get_state(self, site_name: str):
```
Comment (sajith) [inline fabrictestbed_extensions/fablib/resources.py:199]: This needs a docstring.

**Review by sajith — COMMENTED** [review]
Have left some inline comments, mostly as notes to myself.  The plan was to get this PR merged first, and get fixes in with some follow-up PRs.

But turns out that I can't approve this PR, since I happen to be an author. :-/

**Comment by sajith** [comment]
The "check for signed commits" action ran on this branch, although that part of the workflow is not on this branch yet. I am not sure if that is how it is supposed to work -- maybe it is?

**Review by kthare10 — APPROVED** [review]
+1

**Pull request merged**